### DISCUSSION TITLE
Two line intersection cauluation

### DISCUSSION THREAD

**Discussion by zxr201** [NetTopologySuite/NetTopologySuite#503] [Q&A] [answered]
Hello,
We are trying to use NTS to do some GIS calcuation with the data source from AutoCAD. The coordinates of the geometry may be very large, about 1E10 to 1E20 (in our case is 88452586874.108078). When we try to find the minimum rectangle of this geometry, it will get result with displacement. 
We trace the data, turns out it is because in function NetTopologySuite.Algorithm:IntersectionComputer.Intersection, two big double number will loss the precision,

Is there any way to resolve this problem?

Thank you very much.

**Comment by FObermaier** [comment, marked as answer]
You can try the `CommonBitsOp` class:
```C#
var cb = new NetTopologySuite.Precision.CommonBitsOp();
var resGeom = op.Intersection(geom1, geom2);
```
It translates the geometries to compute the intersection of to an origin common to both.
After computing the intersection, the result is translated back to the original origin.

**Reply by zxr201** [reply]
Thank you so much. It seems workd but it still has problem.

My code is like this:
```c#
var r1 = geometryFactory.CreateLineString(pts);
var _cbr = new CommonBitsRemover();
_cbr.Add(r1);
var geom = _cbr.RemoveCommonBits((Geometry)r1.Copy());
var rectangle1 = MinimumDiameter.GetMinimumRectangle(geom);
_cbr.AddCommonBits(rectangle1);
```



When doing "AddCommonBits", it goes to LineString:Apply() to move the coordinates back to the original place. However, at the fisrt point, both the first point and the last point change the cooridnates and when it loops to the last point, the cooridnates changed again. It seems the minimum rectangle is not correct.
I am not sure if RemoveCommonBits and AddCommonBits can be used for all process of geometry.

But still thank you very much. It gives us some ideas that we can try to translate our data first.
